source/MaterialXRuntime/RtStage.cpp

```diff
@@ -171,15 +171,22 @@ RtPrim RtStage::createNodeDef(RtNodeGraph& nodeGraph,
         nodedef.setNodeGroup(nodeGroup);
     }
 
+    // Add an output per nodegraph input
+    for (auto input : nodeGraph.getInputs())
+    {
+        RtAttribute attr = nodedef.createInput(input.getName(), input.getType());
+        attr.setValue(input.getValue());
+    }
+
     // Add an output per nodegraph output
     for (auto output : nodeGraph.getOutputs())
     {
         RtAttribute attr = nodedef.createOutput(output.getName(), output.getType());
         attr.setValue(output.getValue());
     }
 
-    // Set up relationship between nodegraph and nodedef
-    nodeGraph.setNodeDef(prim->hnd());
+    // Set up definition on nodegraph
```

Comment:
This RtRelationship is never used or maintained. Just use what is set on load of a nodedef which is the "nodedef" metadata.